### PR TITLE
Add per degree overview rows on semester overview. Closes 671

### LINKED ISSUE #671: Overview per degree on staff pages
The statistical overview on top of the staff's semester pages should not only show the total values, but also the values per degree.
A table with one line per degree and a total line is suggested. Also the panel should be collapsible like contributor questionnaires on the student's evaluating pages. Whether or not the panel is shown in a collapsed state should be saved in a similar way as the decision whether reviewed comments are displayed on the comment review pages.

Current overview:
overview

### PR DESCRIPTION
**Pull request by simonkrogmann**
Fixes #671

### CODE REVIEW

File: evap/staff/views.py [hunk 1]
```diff
@@ -80,24 +81,37 @@ def semester_view(request, semester_id):
...
+                    stats.num_enrollments_in_evaluation += course.num_participants
+                    stats.num_votes += course.num_voters
+                    stats.num_comments += course.num_textanswers
+                    stats.num_comments_reviewed += course.num_reviewed_textanswers
+                if course.state in ['evaluated', 'reviewed', 'published']:
+                    stats.num_courses_evaluated += 1
+                stats.num_courses += 1
+                stats.first_start = min(stats.first_start, course.vote_start_date)
+                stats.last_end = max(stats.last_end, course.vote_end_date)
+    degree_stats = OrderedDict(sorted(degree_stats.items(), key=lambda x: str(x[0])))
```
Comment (janno42) [inline evap/staff/views.py]: can you please use the `order` attribute of `Degree`?

File: evap/staff/views.py [hunk 2]
```diff
@@ -80,24 +81,37 @@ def semester_view(request, semester_id):
...
-            num_votes += course.num_voters
-            num_comments += course.num_textanswers
-            num_comments_reviewed += course.num_reviewed_textanswers
-        if course.state in ['evaluated', 'reviewed', 'published']:
-            num_courses_evaluated += 1
-        first_start = min(first_start, course.vote_start_date)
-        last_end = max(last_end, course.vote_end_date)
+        for degree in course.degrees.all():
+            if degree not in degree_stats:
+                degree_stats[degree] = Stats()
```
Comment (karyon) [inline evap/staff/views.py]: you could do `degree_stats = defaultdict(Stats)` and then remove this if here.

File: evap/staff/templates/staff_semester_view.html [hunk 3]
```diff
@@ -23,31 +23,39 @@
     {% if num_courses > 0 %}
     <div class="panel panel-info">
         <div class="panel-heading">
-            <h3 class="panel-title">{% trans "Overview" %}</h3>
+            <a class="accordion-toggle" data-toggle="collapse" href="#accordion" onclick="saveCollapseState()">
+                <h3 class="panel-title">{% trans "Overview" %}</h3>
```
Comment (janno42) [inline evap/staff/templates/staff_semester_view.html]: please allow collapsing when clicking on the arrow as well (have a look at the FAQ pages)

**Comment by janno42** [comment]
Courses can have multiple degrees. Please make sure that you don't count them multiple times for the total row ;)

**Comment by janno42** [comment]
After a recent change single results now also have a number of participants. This currently counts towards the total participation, but it shouldn't. Can you please remove the single results from the statistics?

**Comment by janno42** [comment]
@simonkrogmann do you want to continue with this in the next 14 days? if not, i'd like to take over from here.

**Comment by simonkrogmann** [comment]
I will continue working on this feature next week.

**Comment by simonkrogmann** [comment]
Done

**Comment by janno42** [comment]
:+1:
@karyon please review as well

**Comment by karyon** [comment]
thanks!

**Pull request merged**